app/routes/users/users-router.js

```diff
@@ -11,10 +11,10 @@ const raw_parser = body_parser.raw({ type: "image/*", limit: "1mb" });
 user_router.param("user", users_route_param_handlers.user_param_handler);
 
 user_router.all("/all/:email?", middlewares.restrict_to_admin, json_parser, users_controller.get_all_users);
-user_router.get("/:user", middlewares.restrict_to_self, json_parser, users_controller.get_user);
-user_router.post("/", middlewares.restrict_to_self, json_parser, users_controller.create_user);
-user_router.post("/:user/photo", middlewares.restrict_to_self, raw_parser, users_controller.upload_user_photo);
-user_router.patch("/:user", middlewares.restrict_to_self, json_parser, users_controller.update_user);
-user_router.delete("/:user", middlewares.restrict_to_self, json_parser, users_controller.delete_user);
+user_router.get("/:user", middlewares.validate_user, middlewares.restrict_to_self, json_parser, users_controller.get_user);
+user_router.post("/", json_parser, users_controller.create_user);
```

Comment:
firebase uzerinde sign-up basarili ise bize istek atiyor uygulama yani burada da kontrolleri yapmak gerekli diye dusunuyorum. yoksa binlerce iceriye user olusturulabilir gibi geldi.

Reply:
POST user'da firebase token geliyorsa dogru diyorsun, onu duzelteyim.

Reply:
Tabii burada soyle bir durum da var, database'e user henuz kaydolmadigi icin yalnizca token validation yapacagiz. O zaman da validate_user'da https://github.com/gtubt/gtubt-backend/pull/14#discussion_r615374960 burada dedigin gibi invalid tokende 401 dondukten sonra next demememiz lazim.

Reply:
middleware icerisinde next'inin ne oldugu veya endpoint'in ne olduguna bakabiliyor muyuz? simdilik bu endpoint ozelinde bir lojik koyabiliriz? sevmedim mantigimi ama baska bir sey gelmedi simdilik. veya sirf buna ozel bir middleware...